tests/executables_src/testLMapMathPluginPushPars.cc

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: Deutsches Elektronen-Synchrotron DESY, MSK, ChimeraTK Project <chimeratk-support@desy.de>
+// SPDX-License-Identifier: LGPL-3.0-or-later
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE LMapMathPluginTest
+#include <boost/test/unit_test.hpp>
+using namespace boost::unit_test_framework;
+
+#include "Device.h"
+#include "LogicalNameMappingBackend.h"
+
+using namespace ChimeraTK;
+
+BOOST_AUTO_TEST_SUITE(LMapMathPluginTestSuite)
+
+/********************************************************************************************************************/
+
+struct DummyForCleanupCheck : public LogicalNameMappingBackend {
+  using LogicalNameMappingBackend::LogicalNameMappingBackend;
+
+  static boost::shared_ptr<DeviceBackend> createInstance(std::string, std::map<std::string, std::string>) {
+    return boost::make_shared<DummyForCleanupCheck>("mathPluginWithPushPars.xlmap");
+  }
+  ~DummyForCleanupCheck() override {
+    std::cout << "~DummyForCleanupCheck()" << std::endl;
+    cleanupCalled = true;
+  }
+
+  struct BackendRegisterer {
+    BackendRegisterer() {
+      ChimeraTK::BackendFactory::getInstance().registerBackendType(
+          "DummyForCleanupCheck", &DummyForCleanupCheck::createInstance, {"map"});
+    }
+  };
+  static std::atomic_bool cleanupCalled;
+};
+std::atomic_bool DummyForCleanupCheck::cleanupCalled;
+static DummyForCleanupCheck::BackendRegisterer gDFCCRegisterer;
+
+BOOST_AUTO_TEST_CASE(testPushPars) {
+  setDMapFilePath("mathPluginWithPushPars.dmap");
+  {
+    // we enforce that initial values from variable definition in xlmap are never used in MathPugin
+    // - therefore, in test here, we must issue write for all push-parameters
+    // - then, version number check in MathPlugin will notice valid data has been provided after open
+    // - for poll-type accessors, version number check also succeeds since they get recent version numbers anyway
+    // Test needs to call activateAsyncRead but it must be irrelevant whether before or after writes
+
+    ChimeraTK::Device targetDevice;
+    targetDevice.open("HOLD");
+
+    auto accTarget = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/WORD_G");
+    auto pollPar = targetDevice.getScalarRegisterAccessor<uint32_t>("HOLD0/POLLPAR");
+    pollPar = 1;
+    pollPar.write();
+
+    ChimeraTK::Device device;
```

Comment:
Why do you use two devices at once? This is confusing.

Reply:
OK, I get it now. One is the "raw device", i.e. without logical name mapping, the other one is the LMAP-Device. Please make this clear in the names and/or add comments.

Reply:
Right. Renamed it -> logicalDevice